packages/main/src/plugin/provider-registry.ts

```diff
@@ -399,8 +399,22 @@ export class ProviderRegistry {
     // grab the provider
     const provider = this.getMatchingProvider(internalId);
 
-    await autoStart.start(new LoggerImpl());
-
+    let updateContainerConnectionCalled: boolean = false;
+    const context = {
+      updateContainerConnection: (containerProviderConnection: ContainerProviderConnection): void => {
+        updateContainerConnectionCalled = true;
+        const providerConnectionInfo = this.getProviderConnectionInfo(containerProviderConnection);
```

Comment:
question: should we check that this object is matching one being registered by this provider id ? before handling it

to be sure the extension is not giving us another object from other extension/whatever ?